### PR TITLE
Remove Unnecessary Escaping for Tags

### PR DESCRIPTION
**Pull request by jimmychung98**
Jira: https://jira.duraspace.org/browse/ISLANDORA-1750
# **What does this Pull Request do?**

Islandora XML Forms Introduces unnecessary escaping for the tag field.

1) This PR fixes that unnecessary escaping. 
2) Also fixes MODS record in the process because it originally escaped twice too. 
# **How should this be tested?**

Steps to Reproduce:

1) Islandora edits an Object's MODS Datastream. 
2) Change a Subject Topic to something with an ampersand. 
3) Save the record. 
4) Edit the record again.  The ampersand is now HTML encoded as &amp;  Subsequent saves will double encode the character.

image

image

We were unable to find a valid need for the check_plain call. Can you explain if there was something we missed?

### CODE REVIEW

**Comment by whikloj** [comment]
Hi @dra2gon3,

Thanks for the contribution. My guess is that this was to avoid people inputing invalid characters, but I'm not sure.

Either way, could I ask you to open a [JIRA issue](https://jira.duraspace.org/projects/ISLANDORA) where you can describe the issue you are experiencing more fully and can I also point out our [contributing guidelines](https://github.com/Islandora/islandora_xml_forms/blob/7.x/CONTRIBUTING.md). The most important being we will need a contributor license agreement from you.

**Comment by DiegoPino** [comment]
@dra2gon3 you are right, check_plain() should not be there, not sure but FAPI should already handling that. Just in case, can you check if your resulting XML is fine and/or with other strange non html entities? I just want to be sure it's not breaking XML somewhere.

**Comment by jimmychung98** [comment]
Yep, the XML is producing the right results. The only problem seems to be when editing MODS in the datastream.

**Comment by ctgraham** [comment]
@whikloj, the University of Pittsburgh's legal office is currently reviewing the CCLA.  We hope to have that executed soon and that will cover @dra2gon3's work here.

The contributing guidelines provide a link for signing up for a JIRA account, but that JIRA site prevents that.

> Sorry, you can't sign up to this JIRA site at the moment as it's private.

Would you be a good contact for gaining access to JIRA?

**Comment by manez** [comment]
@ctgraham drop me a line at manez@islandora.ca and I will give you a token to sign up for JIRA. They must be in spambot-lockdown.

**Comment by jordandukart** [comment]
Looks like this was introduced here: https://github.com/Islandora/islandora_xml_forms/pull/184 to deal with https://jira.duraspace.org/browse/ISLANDORA-872.

**Comment by dra2gon3** [comment]
I made a new change which fixes the previous and current issues regarding saving special characters into the tag.

**Comment by DonRichards** [comment]
@dra2gon3 as a reminder. Using the Pull Request(PR) template help us quickly look over and test a PR. If you wouldn't mind adding the missing fields from the PULL_REQUEST_TEMPLATE.md file? It would help a lot. 

https://github.com/Islandora/islandora_xml_forms/blob/7.x/.github/PULL_REQUEST_TEMPLATE.md

**Comment by jimmychung98** [comment]
@DonRichards Sorry about that. I updated my original comment and let me know if there is anything else  missing.

**Comment by dannylamb** [comment]
@Islandora/7-x-1-x-committers  Can someone verify that this works and doesn't re-introduce the issue from https://jira.duraspace.org/browse/ISLANDORA-872?

**Comment by ctgraham** [comment]
We received the updated cCLA and have forwarded it for execution internally.  I'll followup to see where that stands with our processes.

**Comment by DiegoPino** [comment]
@ctgraham hi, we are on release time here 7.x-1.9 soon) and wondering if you could make that cCLA happen? Would love to be able to merge this fix. Thanks!

**Comment by ctgraham** [comment]
@DiegoPino, can you give me a desired date?  The cCLA is in review with our legal folks, and I'll use your timeline as leverage to press for a conclusion to this long process.

**Comment by DiegoPino** [comment]
@ctgraham cool. Code Freeze is March 1st. Would be extremely nice if we could have it by Friday 24th, or Monday 27th. I will be adding Release tags/branches March 1st/ 23:59 😬 but the CLA needs to be reviewed by @manez before I merge, so in any case needs to be done before 4:00 PM Atlantic(Canada) time that day as last chance scenario for this release.
Thanks a lot for answering so fast back, really appreciate it!

**Comment by manez** [comment]
Ping @ctgraham any chance that cCLA is close? Code Freeze is coming up tomorrow.

**Comment by ctgraham** [comment]
@manez, I asked Legal for this with a "deadline" of the 24th, but didn't get a response.  I will press the question again, but I think we should expect that we will not make this code freeze.  Sorry.

**Comment by manez** [comment]
@ctgraham @DiegoPino cCLA now on file!

**Comment by DiegoPino** [comment]
WUHU! @ctgraham amazing and congrats for your first pull request that will get merged and be part of our new release, and all this against time!Thanks @manez for making all this possible. 😼 
Merging this now. Good work team! And thanks @dra2gon3 for the actual coding. Almost forget you!

**Review by DiegoPino — APPROVED** [review]

**Pull request merged**
